File: week3-AppDev-crud/crud-nodejs/Ex04_Query5b_TakeOff.js
const connection = require('./db_connection')

// Defining our journey
const journey_id      = UUID.fromString('<enter_your_journey_id>')
const spacecraft_name = 'Crew Dragon Endeavour,SpaceX'

console.log("========================================")
console.log("Start exercise")
console.log("9..8..7..6..5..4..3..2..1 Ignition")

connection.client.execute(
    'UPDATE spacecraft_journey_catalog SET active=true, start=? WHERE spacecraft_name=? AND journey_id=?',
    [ Date.now(), spacecraft_name, journey_id],
    { prepare : true }
)
.then(function (result){
	console.log("Journey %s has now taken off", journey_id.toString());
    connection.client.shutdown()
    console.log("SUCCESS")
})
.catch(function (error){
     console.log(error.message)
     connection.client.shutdown()
});

console.log("========================================")

File: week3-AppDev-crud/crud-python/Ex04_Query5b_TakeOff.py
#!/usr/bin/env python3
from db_connection import Connection
import uuid
import datetime

#Defining our journey
journey_id = uuid.UUID('230995ee-c697-11ea-b7a1-8c85907c08dd')
spacecraft_name = 'Crew Dragon Endeavour,SpaceX'

print("========================================")
print("Start exercise")
print("9..8..7..6..5..4..3..2..1 Ignition")

# this is a update statement in python
try:
    connection = Connection()
    connection.session.execute(
        "UPDATE spacecraft_journey_catalog SET active=true, start= %s WHERE spacecraft_name= %s AND journey_id= %s",
        [datetime.datetime.now(), spacecraft_name, journey_id ])
except Exception as e: 
    print(e)
    print('Failure')
else:
    print("Journey {} has now taken off".format(str(journey_id)))
    print('Success')
finally:
    connection.close()

print('========================================')    